climada/hazard/low_flow.py

```diff
@@ -203,70 +203,119 @@ def set_from_nc(self, input_dir=None, centroids=None, countries=[], reg=None,
         self.tag = TagHazard(haz_type=HAZ_TYPE, file_name=\
-                             FILENAME_NC % (gh_model, cl_model, "*", scenario, soc, \
-                                            fn_str_var, "*_*.nc"), \
+                             f'{gh_model}_{cl_model}_*_{scenario}_{soc}_{fn_str_var}_*.nc', \
                              description='yearrange: %i-%i (%s, %s), reference: %i-%i (%s, %s)' \
                                  %(yearrange[0], yearrange[-1], scenario, soc, \
                                    yearrange_ref[0], yearrange_ref[-1], \
                                    scenario_ref, soc_ref)
                              )
 
+    def _intensity_loop(self, uni_ev, coord, res_centr, num_centr):
+        """Compute and intensity matrix. For each event, if more than one points of
+        data have the same coordinates, take the sum of days below threshold
+        of these points (duration as accumulated intensity).
+
+        Parameters:
+            uni_ev (list of str): list of unique cluster IDs
+            coord (list): Coordinates as in Centroids.coord
+            res_centr (float): Geographical resolution of centroids
+            num_centroids (int): Number of centroids
+
+        Returns:
+            intensity_mat (sparse.lilmatrix): intensity values as sparse matrix
+        """
+        tree_centr = BallTree(coord, metric='chebyshev')
+        stps = list(np.arange(0, len(uni_ev) - 1, INTENSITY_STEP)) + [len(uni_ev)]
+
+        if len(stps) == 1:
+            intensity_list = [self._intensity_one_cluster(tree_centr, cl_id, res_centr, num_centr)
+                  for cl_id in uni_ev]
+            return sparse.csr_matrix(intensity_list)
+
+        for idx, stp in enumerate(stps[0:-1]):
+            intensity_list = []
+            stp_range = uni_ev[stp:stps[idx+1]] if idx else uni_ev[0:stps[1]]
+            for cl_id in stp_range:
+                intensity_list.append(
+                    self._intensity_one_cluster(tree_centr, cl_id,
+                                                res_centr, num_centr))
+                if not idx:
+                    intensity_mat = sparse.csr_matrix(intensity_list)
+                else:
+                    intensity_mat = sparse.vstack((intensity_mat,
+                                               sparse.csr_matrix(intensity_list)))
+        return intensity_mat
+
+    def _set_dates(self, uni_ev):
+        """Set dates of maximum intensity (date) as well as start and end dates
+        per event
+
+        Parameters:
+            uni_ev (list): list of unique cluster IDs
+        """
+        self.date = np.zeros(uni_ev.size, int)
+        self.date_start = np.zeros(uni_ev.size, int)
+        self.date_end = np.zeros(uni_ev.size, int)
+        for ev_idx, ev_id in enumerate(uni_ev):
+            # set event date to date of maximum intensity (ndays)
+            self.date[ev_idx] = self.data[self.data.cluster_id == ev_id]\
```

Comment:
What happens if there is not matching ev_id for the ev_idx? Can this happen? Should maybe an error be raised then?

Reply:
This should not be possible, since `uni_ev` at this point only contains existing events, all filtering out of events happens earlier on.